prover-service/src/training_wheels/verification_logic.rs

```diff
@@ -1,119 +1,22 @@
-use std::{
-    sync::Arc,
-    time::{SystemTime, UNIX_EPOCH},
-};
-
 use aptos_crypto::poseidon_bn254;
-use aptos_keyless_common::input_processing::{
-    config::CircuitConfig,
-    encoding::{FromB64, JwtHeader, JwtParts, JwtPayload},
-};
+use aptos_keyless_common::input_processing::config::CircuitConfig;
 use aptos_types::{
     jwks::rsa::RSA_JWK, keyless::Claims, transaction::authenticator::EphemeralPublicKey,
 };
 use ark_bn254::Fr;
-use http::StatusCode;
 use jsonwebtoken::{Algorithm, DecodingKey, Validation};
 
-use crate::{
-    api::RequestInput,
-    config::ProverServiceConfig,
-    error::{ErrorWithCode, ThrowCodeOnError},
-    input_processing::{field_parser::FieldParser, types::Input},
-    jwk_fetching,
-};
-use anyhow::{bail, Context, Result};
-
-pub fn check_nonce_consistency(input: &Input, circuit_config: &CircuitConfig) -> Result<()> {
-    let payload_decoded = input.jwt_parts.payload_decoded()?;
-    let payload_struct: JwtPayload = serde_json::from_str(&payload_decoded)?;
-    let computed_nonce = compute_nonce(
-        input.exp_date_secs,
-        &input.epk,
-        input.epk_blinder_fr,
-        circuit_config,
-    )?;
-
-    if computed_nonce.to_string() == payload_struct.nonce {
-        Ok(())
-    } else {
-        bail!("Nonce in JWT is inconsistent with epk, epk_blinder, or expiration date")
-    }
-}
-
-pub fn validate_jwt_sig_and_dates(
-    rqi: &RequestInput,
-    maybe_jwk: Option<&RSA_JWK>,
-    config: &ProverServiceConfig,
-) -> Result<(), ErrorWithCode> {
-    let jwt_parts = JwtParts::from_b64(&rqi.jwt_b64)?;
-
-    let header_decoded = jwt_parts.header_decoded()?;
-    let header_struct: JwtHeader = serde_json::from_str(&header_decoded)?;
-
-    let payload_decoded = jwt_parts.payload_decoded()?;
-    let payload_struct: JwtPayload = serde_json::from_str(&payload_decoded)?;
+use crate::config::ProverServiceConfig;
+use anyhow::Result;
 
-    let jwk = match maybe_jwk {
-        Some(x) => Arc::new(x.clone()),
-        None => jwk_fetching::cached_decoding_key(&payload_struct.iss, &header_struct.kid)
-            .context(format!(
-                "Request has a JWT with an unrecognized JWK: {}",
-                payload_struct.iss
-            ))?,
-    };
-
-    // Check the signature verifies.
+pub fn validate_jwt_sig(jwk: &RSA_JWK, jwt: &str, config: &ProverServiceConfig) -> Result<()> {
     let mut validation = Validation::new(Algorithm::RS256);
     if !config.enable_jwt_exp_not_in_the_past_check {
```

Comment:
Same: Can you please 🙏 clarify when this would ever be disabled? And can you please add it as a comment above this?